### PR TITLE
Fix issue in the copy API

### PR DESCRIPTION
**Pull request by kalaiyarasiganeshalingam**
## Purpose
$Subject 

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2837
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec

### CODE REVIEW

File: ballerina/tests/file-test.bal [hunk 1]
```diff
@@ -77,6 +78,19 @@ function testCopyDir() {
...
+@test:Config {}
+function testCopyDir2() returns error? {
+    string targetPath = "tests/resources/temp-dir/nested-file.txt";
+    error? copyResult = copy(srcDir, "tests/resources/temp-dir", REPLACE_EXISTING);
+    if copyResult is error {
+        test:assertFail("Directory not copied!");
+    } else {
+        string readContent = check io:fileReadString(targetPath);
+        test:assertEquals(readContent, "Hi");
+        check io:fileWriteString(targetPath, "");
```
Comment (daneshk) [inline ballerina/tests/file-test.bal:90]: shall we just remove the file, instead making the content empty

Comment (daneshk) [reply ballerina/tests/file-test.bal:90]: do we need an empty file in the target directory always

Comment (daneshk) [reply ballerina/tests/file-test.bal:90]: Shall we have at least two files in the src dir

File: native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java [hunk 2]
```diff
@@ -304,6 +308,34 @@ public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IO
...
+            this.source = source;
+            this.target = target;
+            this.copyOptions = copyOptions;
+        }
+
+        @Override
+        public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) throws IOException {
+            Path targetPath = target.resolve(source.relativize(dir));
+            if (!Files.exists(targetPath)) {
+                Files.createDirectory(targetPath);
```
Comment (daneshk) [inline native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java:327]: what if parent directories do not exist. Don't we need to use `Files. createDirectories` API to create all the missing directories of the target path

File: ballerina/tests/file-test.bal [hunk 3]
```diff
@@ -77,6 +78,19 @@ function testCopyDir() {
...
+    }
+}
+
```
Comment (daneshk) [inline ballerina/tests/file-test.bal:93]: We need to have a test case where the target directory doesn't exist

Comment (kalaiyarasiganeshalingam) [reply ballerina/tests/file-test.bal:93]: That is already existing:https://github.com/ballerina-platform/module-ballerina-file/blob/master/ballerina/tests/file-test.bal#L73

File: native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java (hunk 2 above)
Comment (kalaiyarasiganeshalingam) [reply native/src/main/java/io/ballerina/stdlib/file/nativeimpl/Utils.java:327]: This does not create all the missing directories. If the parent exists, only a child directory is created by this. The existing API also works like this.

File: ballerina/tests/file-test.bal (hunk 3 above)
Comment (daneshk) [reply ballerina/tests/file-test.bal:93]: Then how come it is passing. is the problem only when we try to replace existing files.

File: ballerina/tests/file-test.bal (hunk 1 above)
Comment (daneshk) [reply ballerina/tests/file-test.bal:90]: This comment is not valid. The issue only happens when we try to replace existing files

**Review by daneshk — APPROVED** [review]

File: ballerina/tests/file-test.bal (hunk 1 above)
Comment (kalaiyarasiganeshalingam) [reply ballerina/tests/file-test.bal:90]: I have added [this test by creating files](https://github.com/ballerina-platform/module-ballerina-file/pull/429/commits/e471c70b7b72d1025056dcfd9e98910c789c8713#diff-66a9555be220b2129eec3c611997479d1e31ecd1f9a93ad74fe37b5a3a3c6cabR83). It was failed in the GitHub workflow, but the local build succeeded.

**Pull request merged**